### PR TITLE
Add hash-git command

### PR DESCRIPTION
**Pull request by matthewbauer**

### CODE REVIEW

File: src/nix/hash.cc [hunk 1]
```diff
@@ -4,12 +4,13 @@
...
 #include "references.hh"
 #include "archive.hh"
+#include "git.hh"
 
 using namespace nix;
 
 struct CmdHash : Command
 {
-    enum Mode { mFile, mPath };
+    enum Mode { mFile, mPath, mGit };
```
Comment (Ericson2314) [inline src/nix/hash.cc:13]: Hmm, I guess we use `FileIngestionMethod` for this now?

Comment (matthewbauer) [reply src/nix/hash.cc:13]: I think it should be separate. These are just modes of the has command.

File: src/nix/hash.cc [hunk 2]
```diff
@@ -55,6 +59,8 @@ struct CmdHash : Command
 
             if (mode == mFile)
```
Comment (Ericson2314) [inline src/nix/hash.cc:60]: And we can use switch-case here

File: src/nix/hash.cc (hunk 1 above)
Comment (Ericson2314) [reply src/nix/hash.cc:13]: I am fine doing in separate PR but I think it is no coincidence both before and after they are 1-1.

**Pull request merged**